### PR TITLE
Fix for issue #1199

### PR DESCRIPTION
**Pull request by polybassa**

### CODE REVIEW

**Comment by guedou** [comment]
Thanks for the PR. Could you provide unit tests?

For example:
```
>>> packet = IP(dst="8.8.8.8")/ICMP() ; r = sr(packet); packet.sent_time is not None
>>> packet = IP(dst="8.8.8.8")/ICMP() ; r = sr1(packet); packet.sent_time is not None
```

**Comment by guedou** [comment]
It was only examples. Please remove the ';' and add assert before the tests for None. Thanks.

**Comment by polybassa** [comment]
Sorry

File: scapy/sendrecv.py [hunk 1]
```diff
@@ -183,7 +183,7 @@ def sndrcv(pks, pkt, timeout=None, inter=0, verbose=None, chainCC=False,
     nbrecv = 0
     ans = []
     # do it here to fix random fields, so that parent and child have the same
-    tobesent = [p for p in pkt]
+    tobesent = [pkt]
```
Comment (gpotter2) [inline scapy/sendrecv.py]: Your first proposal `list` was correct, this one isn’t. A packet may be a generator, a list...

**Comment by karpierz** [comment]
Why not?:
```tobesent = list(pkt) if hasattr(pkt, "__iter__") else pkt```

**Comment by gpotter2** [comment]
I think it's safe to call simply `list(pkt)`:
- if pkt is a list, list() returns the same object
- if pkt is a packet, or a generator, __iter__ is implicitly called

**Comment by polybassa** [comment]
Now I got a problem. If I do the following:
``` tobesent = [pkt] ```
this issue is fixed, but some other unit test will fail.
if I do 
``` tobesent = list(pkt) ```
this issue is still present, but other unit test will not fail.

**Comment by gpotter2** [comment]
This means that the fix is incorrect. this line is more complex that it looks, knowing that a packet can generate others depending on its parameters

**Comment by karpierz** [comment]
> I think it's safe to call simply list(pkt):
>  if pkt is a list, list() returns the same object

It is not quite truet.
list(list_object) returns a new list object (copy of list_object). Try:
a = [1,2,3,4]
b = list(a)
a is b

**Comment by polybassa** [comment]
ok, the problem is probably somewhere inside packet.__iter__

**Comment by polybassa** [comment]
unfortunately, I don't know enough internals of scapy, to fix this issue

**Comment by karpierz** [comment]
I will try to help you.

**Comment by karpierz** [comment]
I will try to help you.
UPDATE: Solved. See at #1282
